project/apps/api-gateway/src/auth/auth.guard.ts

```diff
@@ -17,17 +17,66 @@ export class AuthGuard implements CanActivate {
 
   async canActivate(context: ExecutionContext): Promise<boolean> {
     const request = context.switchToHttp().getRequest();
-    const token = request.cookies['token'];
+    const response = context.switchToHttp().getResponse();
+    const accessToken = request.cookies['access_token'];
+    const refreshToken = request.cookies['refresh_token'];
 
-    if (!token) {
+    if (accessToken) {
+      const { data, error } = await firstValueFrom(
```

Comment:
It should be this instead right? Your endpoint just returns u the data not an error here
```suggestion
      const data = await firstValueFrom(
```